app/waf/src/controllers/schema.ts

```diff
@@ -0,0 +1,284 @@
+
+    s.setStatusCodeExample(400);
+    s.setNameExample('BadRequestError');
+    s.removeCodeProperty();
+    s.removeDetailsProperty();
+
+    return s;
+  }
+
+  public static NotFound(): ErrorResponseSchema {
+    let s = new ErrorResponseSchema();
+
+    s.setStatusCodeExample(404);
+    s.setNameExample('Error');
+    s.setMessageExample(
+      'Entity not found: XXXX with id "11111111-2222-3333-4444-555555555555"',
+    );
+    s.setCodeExample('ENTITY_NOT_FOUND');
+    s.removeDetailsProperty();
+
+    return s;
+  }
+
+  public static UnprocessableEntity(): ErrorResponseSchema {
+    let s = new ErrorResponseSchema();
+
+    s.setStatusCodeExample(422);
+    s.setNameExample('UnprocessableEntityError');
+    s.setMessageExample(
+      'The request body is invalid. See error object `details` property for more info.',
+    );
+    s.setCodeExample('VALIDATION_FAILED');
+
+    return s;
+  }
+
+  public setStatusCodeExample(statusCode: number) {
+    this.schema.schema.properties.statusCode.example = statusCode;
+  }
+
+  public setNameExample(name: string) {
+    this.schema.schema.properties.name.example = name;
+  }
+
+  public setMessageExample(message: string) {
+    this.schema.schema.properties.message.example = message;
+  }
+
+  public setCodeExample(code: string) {
+    this.schema.schema.properties.code.example = code;
+  }
+
+  public addDetailExample(detail: Object) {
+    this.schema.schema.properties.details.example.push(detail);
+  }
+
+  public removeCodeProperty() {
+    delete this.schema.schema.properties.code;
+  }
+
+  public removeDetailsProperty() {
+    delete this.schema.schema.properties.details;
+  }
+}
+
+function buildParameterSchema(
+  name: string,
+  location: ParameterLocation,
+  required: boolean,
+  type: string,
+  format: string,
+  description: string,
+): ParameterObject {
+  return {
+    name: name,
+    description: description,
+    in: location,
+    required: required,
+    schema: {
+      type: type,
+      format: format,
+    },
+  };
+}
+
+function buildRequestSchema(
+  description: string,
+  requiredProperties: string[],
+  properties: Object,
+  example: Object,
+): Object {
+  return {
+    description: description,
+    required: true,
+    content: {
+      'application/json': {
+        schema: {
+          type: 'object',
+          required: requiredProperties,
+          properties: properties,
+          example: example,
+        },
+      },
+    },
+  };
+}
+
+function buildResponseSchema(properties: Object, example: Object): Object {
+  return {
+    schema: {
+      type: 'object',
+      properties: properties,
+      example: example,
+    },
+  };
+}
+
+function buildCollectionResponseSchema(
+  properties: Object,
+  example: Object,
+): Object {
+  return {
+    schema: {
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: properties,
+        example: example,
+      },
+    },
+  };
+}
+
+export class Schema {
```

Comment:
Will we rename the class name from `Schema` to 'AdcSchema', since there may be more schema for all models?

Reply:
OK

Reply:
Done.